microsoft.azure.functions.worker.sdk 2.0.3 fail dotnet publish command

### Description

Under .Net 8.0, microsoft.azure.functions.worker.sdk 2.0.3 caused dotnet publish command to fail. Worked well in 2.0.2.

### Steps to reproduce

Running this command:
`C:\hostedtoolcache\windows\dotnet\dotnet.exe publish some.csproj --configuration debug --output somefolder --no-build`

results below error:
`C:\Users\VssAdministrator\.nuget\packages\microsoft.azure.functions.worker.sdk\2.0.3\build\Microsoft.Azure.Functions.Worker.Sdk.targets(234,5): error MSB4044: The "AssignTargetPath" task was not given a value for the required parameter "RootFolder".`

Same thing happens in all 3 of our .NET 9.0 function projects.
Build and tests work fine, but the the publish command fails.

```
  - task: DotNetCoreCLI@2
    displayName: .NET publish Functions
    inputs:
      command: publish
      publishWebProjects: false
      arguments: '--no-build --configuration Release --output $(Build.ArtifactStagingDirectory)/packages'
      projects: src/redacted/redacted.csproj
      zipAfterPublish: true
```
Error:
`/home/vsts/.nuget/packages/microsoft.azure.functions.worker.sdk/2.0.3/build/Microsoft.Azure.Functions.Worker.Sdk.targets(234,5): error MSB4044: The "AssignTargetPath" task was not given a value for the required parameter "RootFolder" ...`

Downgrading to Microsoft.Azure.Functions.Worker.Sdk 2.0.2 fixes the problem

--- a/sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets
+++ b/sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets
@@ -141,7 +141,7 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
   <!-- For publish we have very little targets. Either:
     1) It is a publish with build, in which case _FunctionsInnerBuild will already be ran.
     2) It is publish no-build, in which case we assume everything is as expected in the output directory already. -->
-  <Target Name="_FunctionsInnerPublish" BeforeTargets="GetCopyToPublishDirectoryItems" DependsOnTargets="_FunctionsExtensionAssignTargetPaths" />
+  <Target Name="_FunctionsInnerPublish" BeforeTargets="GetCopyToPublishDirectoryItems" DependsOnTargets="_FunctionsPreBuild;_WorkerExtensionsBuild;_FunctionsExtensionAssignTargetPaths" />
 
   <!-- Invoke restore on WorkerExtension.csproj.
     We depend on _GetRestoreSettings, which is a task from nuget targets that will resolve various nuget restore settings.
@@ -154,6 +154,7 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
   <Target Name="_WorkerExtensionsBuild">
     <PropertyGroup>
       <_WorkerExtensionTarget>Build</_WorkerExtensionTarget>
+      <_WorkerExtensionTarget Condition="'$(NoBuild)' == 'true'">GetTargetPath</_WorkerExtensionTarget>
       <_WorkerExtensionProperties>Configuration=Release;$(_FunctionsExtensionCommonProps)</_WorkerExtensionProperties>
     </PropertyGroup>
 

--- a/sdk/release_notes.md
+++ b/sdk/release_notes.md
@@ -4,9 +4,9 @@
 - My change description (#PR/#issue)
 -->
 
-### Microsoft.Azure.Functions.Worker.Sdk <version>
+### Microsoft.Azure.Functions.Worker.Sdk 2.0.4
 
-- <entry>
+- Address issue with `dotnet publish --no-build` producing an error. (#3068)
 
 ### Microsoft.Azure.Functions.Worker.Sdk.Generators <version>
 

